anabot/runtime/installation/hub/partitioning/advanced/__init__.py

```diff
@@ -96,7 +96,12 @@ def schema_manipulate(element, app_node, local_node, check):
         return (False, u"Different schema is set: %s" % schema_node.name)
     # Select new schema
     schema_node.click()
-    getnode(schema_node, "menu item", schema_name(schema)).click()
+    try:
+        getnode(schema_node, "menu item", schema_name(schema)).click()
+    except TimeoutError:
+        # If schema name is valid but not found in the dropdown menu (e.g.
+        # 'btfrs' in RHEL), getnode raises TimeoutError.
+        pass
```

Comment:
Yes, it is true it should not crash, however it should not silently continue either. At this point, the dropdown menu would remain to be shown as it was "opened" by the `schema_node.click()` so e.g. ESC key should be pressed to "close" the dropdown menu.

The other problem I see (not introduced directly by this change) is that `schema_check` uses `@check_action_result` however the action `schema_handler` does not return any result that could be used. In this case, instead of just continuing, it would be much better to return failure that such schema is not available (be it valid or invalid - there's no checking whether it's valid or not which is imho correct) and the `schema_handler` should return the return value of `schema_manipulate`.

Reply:
@pholica Thanks for comments. You are right, errors should not pass silently. Added some code to return errors and to handle related exceptions. Can you please have a look whether it seems ok to you?